pkg/ruler/ruler.go

```diff
@@ -63,6 +66,11 @@ type Config struct {
 	SearchPendingFor time.Duration
 	LifecyclerConfig ring.LifecyclerConfig
 	FlushCheckPeriod time.Duration
+
+	EnableAPI bool `yaml:"enable_api"`
+
+	// used to avoid duplicate registration in tests
+	registry prometheus.Registerer `yaml:"-"`
```

Comment:
A better approach would be passing the `prometheus.Registerer` in input to `newManager()` (and propagate it from parents if required). This way tests can pass a custom registerer. It's a common design, we already use in other parts of Cortex.

Reply:
👍